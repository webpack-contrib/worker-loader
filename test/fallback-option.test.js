import path from 'path';

import del from 'del';

import {
  compile,
  getCompiler,
  getErrors,
  // getModuleSource,
  getResultFromBrowser,
  getWarnings,
} from './helpers';

describe('"fallback" option', () => {
  beforeAll(() => del(path.resolve(__dirname, `outputs`)));

  it('should work by default', async () => {
    const compiler = getCompiler('./basic/entry.js', {
      inline: true,
      fallback: true,
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    // expect(getModuleSource('./basic/worker.js', stats)).toMatchSnapshot(
    //   'module'
    // );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should should work with "true" value', async () => {
    const compiler = getCompiler('./basic/entry.js', {
      inline: true,
      fallback: true,
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    // expect(getModuleSource('./basic/worker.js', stats)).toMatchSnapshot(
    //   'module'
    // );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should should work with "false" value', async () => {
    const compiler = getCompiler('./basic/entry.js', {
      inline: true,
      fallback: false,
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    // expect(getModuleSource('./basic/worker.js', stats)).toMatchSnapshot(
    //   'module'
    // );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });
});
