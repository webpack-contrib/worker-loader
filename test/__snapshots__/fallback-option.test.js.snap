// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"fallback" option should should work with "false" value: errors 1`] = `Array []`;

exports[`"fallback" option should should work with "false" value: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"fallback" option should should work with "false" value: warnings 1`] = `Array []`;

exports[`"fallback" option should should work with "true" value: errors 1`] = `Array []`;

exports[`"fallback" option should should work with "true" value: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"fallback" option should should work with "true" value: warnings 1`] = `Array []`;

exports[`"fallback" option should work by default: errors 1`] = `Array []`;

exports[`"fallback" option should work by default: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"fallback" option should work by default: warnings 1`] = `Array []`;
